pkg/js/util/exceptions.ts

```diff
@@ -0,0 +1,117 @@
+import { RecognitionException } from "antlr4";
+
+export enum ValidationError {
```

Comment:
Move these and the public error interfaces to `pkg/js/errors`, keep any internal error interfaces here